src/main/java/com/revealprecision/revealserver/api/v1/facade/service/TaskFacadeService.java

```diff
@@ -50,21 +67,68 @@ public class TaskFacadeService {
   private final PersonService personService;
   private final LocationService locationService;
   private final BusinessStatusService businessStatusService;
+  private final KafkaTemplate<String, Message> kafkaTemplate;
+  private final StreamsBuilderFactoryBean getKafkaStreams;
+  private final KafkaProperties kafkaProperties;
 
   public List<TaskFacade> syncTasks(List<String> planIdentifiers,
-      List<UUID> jurisdictionIdentifiers, Long serverVersion) {
-
-    log.info("Before task sync");
-    List<TaskFacade> collect = planIdentifiers.stream().map(
-            planIdentifier -> taskService.getTasksPerJurisdictionIdentifier(
-                    UUID.fromString(planIdentifier), jurisdictionIdentifiers, serverVersion).entrySet()
-                .stream()
-                .map(this::getTaskFacades).flatMap(Collection::stream).collect(Collectors.toList()))
-        .flatMap(Collection::stream).collect(Collectors.toList());
-    log.info("Done with task sync");
-    return collect;
+      List<UUID> jurisdictionIdentifiers, Long serverVersion, String requester) {
+
+    List<Location> operational = jurisdictionIdentifiers.stream().map(jurisdictionIdentifier -> {
+      Location location = locationService.findByIdentifier(jurisdictionIdentifier);
+      return location;
+    }).filter(location -> location.getGeographicLevel().getName().equals("operational")).collect(
+        Collectors.toList());
+    log.debug("getting tasks for the ffg operational areas: {}",operational);
+
+    KafkaStreams kafkaStreams = getKafkaStreams.getKafkaStreams();
+    ReadOnlyKeyValueStore<String, TaskEvent> taskPlanParent = kafkaStreams.store(
+        StoreQueryParameters.fromNameAndType(kafkaProperties.getStoreMap().get(KafkaConstants.taskPlanParent),
+            QueryableStoreTypes.keyValueStore())
+    );
+    ReadOnlyKeyValueStore<String, TaskAggregate> taskParent = kafkaStreams.store(
+        StoreQueryParameters.fromNameAndType(kafkaProperties.getStoreMap().get(KafkaConstants.taskParent),
+            QueryableStoreTypes.keyValueStore())
+    );
+
+    log.debug("Before task sync");
+
+    Set<String> strings = planIdentifiers.stream()
+        .flatMap(
+            planIdentifier -> operational.stream()
+                .peek(plan->log.debug("plan Id for task sync: {}",planIdentifier))
+                .flatMap(
+                jurisdictionIdentifier -> {
+                  String taskKey = planIdentifier + "_" + jurisdictionIdentifier.getIdentifier();
+                  log.debug("key to retrieve task: {}",taskKey);
+                  List<TaskLocationPair> taskIds = new ArrayList<>();
+                  try {
+                    taskIds = taskParent.get(taskKey).getTaskIds();
```

Comment:
this assumes the kafka store will never move? else we getting error 500 here.If operationally we are taking care of data store then this is fine.